app/api/v1/access.py

```
@@ -0,0 +1,183 @@
+        )
+
+    except (VaultNotFoundError, UserNotFoundError) as e:
+        print(f"Error: {e}")
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Resource not found")
+    except (UnauthorizedVaultAccessError, InsufficientPermissionsError) as e:
+        print(f"Error: {e}")
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Forbidden")
+    except ValueError as e:
+        # e.g., attempting to add the owner as a member
+        print(f"Error: {e}")
+        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail=str(e))
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)